example/advection/advection_package.cpp

```diff
@@ -276,6 +298,50 @@ void PostFill(MeshBlockData<Real> *rc) {
       });
 }
 
+// Example of how to enroll a history function.
+// Templating is *NOT* required and just implemented here to reuse this function
+// for testing of the UserHistoryOperations curently available in Parthenon (Sum, Min,
+// Max), which translate to the MPI reduction being called over all ranks. T should be
+// either Kokkos::Sum, Kokkos::Min, or Kokkos::Max.
+template <typename T>
+Real AdvectionHst(MeshData<Real> *md) {
+  auto hydro_pkg =
+      md->GetBlockData(0)->GetBlockPointer()->packages.Get("advection_package");
+
+  // Packing variable over MeshBlock as the function is called for MeshData, i.e., a
+  // collection of blocks
+  const auto &advected_pack = md->PackVariables(std::vector<std::string>{"advected"});
+
+  IndexRange ib = advected_pack.cellbounds.GetBoundsI(IndexDomain::interior);
+  IndexRange jb = advected_pack.cellbounds.GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = advected_pack.cellbounds.GetBoundsK(IndexDomain::interior);
+
+  Real result = 0.0;
+  T reducer(result);
+
+  // We choose to apply volume weighting when using the sum reduction.
+  // Downstream this choice will be done on a variable by variable basis and volume
+  // weighting needs to be applied in the reduction region.
+  bool volume_weighting = std::is_same<T, Kokkos::Sum<Real, DevExecSpace>>::value;
+
+  Kokkos::parallel_reduce(
```

Comment:
why not use `par_reduce` since we have it?

Reply:
Because it's slow, see #466 ;) I updated to par_reduce here and will fix `par_reduce` in a separate PR.